### PR TITLE
Feature hyper

### PR DESCRIPTION
**Pull request by planes**
This is not complete yet. There is still more changes to combustor and others to streamline the flow.

### CODE REVIEW

**Comment by planes** [comment]
Last update addresses #187

File: trunk/SUAVE/Components/Energy/Converters/Combustor.py [hunk 1]
```diff
@@ -112,28 +121,139 @@ def compute(self,conditions):
...
 
         # method to compute combustor properties
 
         # method - computing the stagnation enthalpies from stagnation temperatures
         ht4     = Cp*Tt4
         ho      = Cp*To
         ht_in   = Cp*Tt_in
-
+        
         # Using the Turbine exit temperature, the fuel properties and freestream temperature to compute the fuel to air ratio f
```
Comment (WalterMaier) [inline trunk/SUAVE/Components/Energy/Converters/Combustor.py]: Turbine exit temp?  No turbine

File: trunk/SUAVE/Components/Energy/Converters/Combustor.py [hunk 2]
```diff
@@ -112,28 +121,139 @@ def compute(self,conditions):
...
         ht_out  = Cp*Tt4
-        Pt_out  = Pt_in*pib
         
         # pack computed quantities into outputs
         self.outputs.stagnation_temperature  = Tt4
         self.outputs.stagnation_pressure     = Pt_out
         self.outputs.stagnation_enthalpy     = ht_out
         self.outputs.fuel_to_air_ratio       = f 
     
+    def compute_rayleigh(self,conditions):
```
Comment (WalterMaier) [inline trunk/SUAVE/Components/Energy/Converters/Combustor.py]: This may belong in the Methods sections

File: trunk/SUAVE/Components/Energy/Converters/Combustor.py [hunk 3]
```diff
@@ -112,28 +121,139 @@ def compute(self,conditions):
...
+        Mach[i_rayleigh]    = fm_solver(ar,Mach[i_rayleigh],gamma)  
+        
+        # Determine max stagnation temperature to thermally choke flow                                     
+        Tt4_ray = Tt_in*(1+gamma*Mach**2)**2/((2*(1+gamma)*Mach**2)*(1+(gamma-1)/2*Mach**2)) 
+
+        # Checking if Tt4 is limited by Rayleigh
+        i_low = Tt4_ray <= Tt4
+        i_high = Tt4_ray > Tt4
+        
+        # Choose Tt4 for fuel calculations
```
Comment (WalterMaier) [inline trunk/SUAVE/Components/Energy/Converters/Combustor.py]: As an input?

File: trunk/SUAVE/Components/Energy/Converters/Combustor.py [hunk 4]
```diff
@@ -112,28 +121,139 @@ def compute(self,conditions):
...
+        
+        # --Material limitations define Tt4
```
Comment (WalterMaier) [inline trunk/SUAVE/Components/Energy/Converters/Combustor.py]: What does this have to do with materials?  Tt4=Tt4?

File: trunk/SUAVE/Components/Energy/Converters/Combustor.py [hunk 5]
```diff
@@ -112,28 +121,139 @@ def compute(self,conditions):
...
+        
+        # Isentropic decceleration through divergent nozzle
+        Mach[i_rayleigh]    = fm_solver(ar,Mach[i_rayleigh],gamma)  
+        
+        # Determine max stagnation temperature to thermally choke flow                                     
+        Tt4_ray = Tt_in*(1+gamma*Mach**2)**2/((2*(1+gamma)*Mach**2)*(1+(gamma-1)/2*Mach**2)) 
+
+        # Checking if Tt4 is limited by Rayleigh
+        i_low = Tt4_ray <= Tt4
+        i_high = Tt4_ray > Tt4
```
Comment (WalterMaier) [inline trunk/SUAVE/Components/Energy/Converters/Combustor.py]: Used?

File: trunk/SUAVE/Components/Energy/Converters/Combustor.py [hunk 6]
```diff
@@ -112,28 +121,139 @@ def compute(self,conditions):
...
         # unpacking values from self
-        htf    = self.fuel_data.specific_energy        
+        htf             = self.fuel_data.specific_energy
+        ar              = self.area_ratio
+        
+        # compute pressure
+        Pt_out      = Pt_in*pib
+
 
         # method to compute combustor properties
```
Comment (WalterMaier) [inline trunk/SUAVE/Components/Energy/Converters/Combustor.py]: Used?

File: trunk/SUAVE/Components/Energy/Converters/Combustor.py (hunk 5 above)
Comment (planes) [reply trunk/SUAVE/Components/Energy/Converters/Combustor.py]: You're right i_high is unnecessary. I've deleted it.

File: trunk/SUAVE/Components/Energy/Converters/Combustor.py (hunk 4 above)
Comment (planes) [reply trunk/SUAVE/Components/Energy/Converters/Combustor.py]: He's just making an array. I've made it explicit.

File: trunk/SUAVE/Components/Energy/Converters/Combustor.py (hunk 3 above)
Comment (planes) [reply trunk/SUAVE/Components/Energy/Converters/Combustor.py]: I don't know. Please add your changes to make it happen.

Comment (WalterMaier) [reply trunk/SUAVE/Components/Energy/Converters/Combustor.py]: Not exactly sure how to change/what it is doing.

File: regression/scripts/ramjet_network/ramjet_network.py [hunk 7]
```diff
@@ -0,0 +1,228 @@
...
+    
+    #error data function
+    error =  Data()
+    
+    error.thrust_error = (F[0][0] -  expected.thrust)/expected.thrust
+    error.mdot_error   = (mdot[0][0] - expected.mdot)/expected.mdot
+    print error
+    
+    for k,v in error.items():
+        assert(np.abs(v)<1e-4)
```
Comment (timdmacdo) [inline regression/scripts/ramjet_network/ramjet_network.py]: Any tolerance used greater than 1e-6 needs to be justified and reason should be recorded in a comment.

File: trunk/SUAVE/Components/Energy/Converters/Combustor.py [hunk 8]
```diff
@@ -112,28 +121,135 @@ def compute(self,conditions):
...
+        
+        # unpacking values from self
+        htf             = self.fuel_data.specific_energy
+        ar              = self.area_ratio
+        
+        # Rayleigh flow analysis, constant pressure burner
+            
+        # Initialize arrays
+        M_out  = 1*Pt_in/Pt_in
+        Ptr   = 1*Pt_in/Pt_in
```
Comment (timdmacdo) [inline trunk/SUAVE/Components/Energy/Converters/Combustor.py]: spacing

File: trunk/SUAVE/Components/Energy/Converters/Compression_Nozzle.py [hunk 9]
```diff
@@ -98,38 +98,64 @@ def compute(self,conditions):
...
+            Pt_out   = 1.0 * Pt_in/Pt_in
+            P_out    = 1.0 * Pt_in/Pt_in
+            
+            #-- Inlet Mach <= 1.0, isentropic relations
+            Pt_out[i_low]  = Pt_in[i_low]*pid
+            Mach[i_low]    = np.sqrt( (((Pt_out[i_low]/Po[i_low])**((gamma-1.)/gamma))-1.) *2./(gamma-1.) )
+            T_out[i_low]   = Tt_out[i_low]/(1+(gamma-1)/2*Mach[i_low]*Mach[i_low])
+            
+            #-- Inlet Mach > 1.0, normal shock
+            Mach[i_high]   = np.sqrt((1.+(gamma-1.)/2.*Mo[i_high]**2.)/(gamma*Mo[i_high]**2-(gamma-1.)/2.))
```
Comment (timdmacdo) [inline trunk/SUAVE/Components/Energy/Converters/Compression_Nozzle.py]: Squared values should be x*x for code speed reasons.

File: trunk/SUAVE/Components/Energy/Converters/Supersonic_Nozzle.py [hunk 10]
```diff
@@ -169,7 +174,154 @@ def compute(self,conditions):
...
+        
+        #-- Subsonic and sonic flow
+        P_out[i_sub]        = Po[i_sub]
+        M_out[i_sub]        = np.sqrt((((Pt_out[i_sub]/P_out[i_sub])**((gamma-1)/gamma))-1)*2/(gamma-1))
+        A_ratio[i_sub]      = 1./fm_id(M_out[i_sub],gamma)
+        
+        #-- Shock inside nozzle
+        P_out[i_shock]      = Po[i_shock]
+        M_out[i_shock]      = np.sqrt((((Pt_out[i_shock]/P_out[i_shock])**((gamma-1)/gamma))-1)*2/(gamma-1))
+        #M_out[i_shock]      = exit_Mach_shock(A_ratio[i_shock], gamma, Pt_out[i_shock], Po[i_shock])
```
Comment (timdmacdo) [inline trunk/SUAVE/Components/Energy/Converters/Supersonic_Nozzle.py]: should this be removed?

File: trunk/SUAVE/Components/Energy/Networks/Ramjet.py [hunk 11]
```diff
@@ -0,0 +1,242 @@
...
+			exit_velocity                            [m/s] 
+		    fan:
+			exit_static_temperature                  [K]  
+			exit_static_pressure                     [K] 
+			exit_stagnation_temperature              [K] 
+			exit_stagnation_pressure                 [Pa] 
+			exit_velocity                            [m/s] 
+    
+		Properties Used:
+		Defaulted values
```
Comment (timdmacdo) [inline trunk/SUAVE/Components/Energy/Networks/Ramjet.py]: this should be explicit

File: trunk/SUAVE/Components/Energy/Networks/Ramjet.py [hunk 12]
```diff
@@ -0,0 +1,242 @@
...
+        thrust.inputs.core_nozzle                              = core_nozzle.outputs
+	
+        #link the thrust component to the combustor
+        thrust.inputs.fuel_to_air_ratio                        = combustor.outputs.fuel_to_air_ratio
+	
+        #link the thrust component to the low pressure compressor 
+        thrust.inputs.stag_temp_lpt_exit                       = core_nozzle.outputs.stagnation_temperature
+        thrust.inputs.stag_press_lpt_exit                      = core_nozzle.outputs.stagnation_pressure
+        thrust.inputs.number_of_engines                        = number_of_engines
+	thrust.inputs.flow_through_core                        =  1.0 #scaled constant to turn on core thrust computation
```
Comment (timdmacdo) [inline trunk/SUAVE/Components/Energy/Networks/Ramjet.py]: indentation

File: trunk/SUAVE/Methods/Propulsion/fm_id.py [hunk 13]
```diff
@@ -1,33 +1,31 @@
...
 #  fm_id
 # ----------------------------------------------------------------------
 
 ## @ingroup Methods-Propulsion
-def fm_id(M):
-    """
-    Function that takes in the Mach number, and outputs a function fm
-    that's commonly used in compressible flow calculations
-    
+def fm_id(M,gamma):
```
Comment (timdmacdo) [inline trunk/SUAVE/Methods/Propulsion/fm_id.py]: This needs a source.

File: trunk/SUAVE/Methods/Propulsion/fm_solver.py [hunk 14]
```diff
@@ -0,0 +1,52 @@
...
+
+# ----------------------------------------------------------------------
+#  fm_solver
+# ----------------------------------------------------------------------
+
+## @ingroup Methods-Propulsion
+
+
+def fm_solver(area_ratio, M0, gamma):
+    """Function that takes in an area ratio and a Mach number associated to
```
Comment (timdmacdo) [inline trunk/SUAVE/Methods/Propulsion/fm_solver.py]: Needs a source

File: trunk/SUAVE/Methods/Propulsion/nozzle_calculations.py [hunk 15]
```diff
@@ -0,0 +1,153 @@
+## @ingroup Methods-Propulsion
```
Comment (timdmacdo) [inline trunk/SUAVE/Methods/Propulsion/nozzle_calculations.py]: Sources should be found for these functions.

File: trunk/SUAVE/Methods/Propulsion/ramjet_sizing.py [hunk 16]
```diff
@@ -0,0 +1,167 @@
...
+            conditions = SUAVE.Analyses.Mission.Segments.Conditions.Aerodynamics()            
+        
+            # freestream conditions    
+            conditions.freestream.altitude           = np.atleast_1d(altitude)
+            conditions.freestream.mach_number        = np.atleast_1d(mach_number)
+            conditions.freestream.pressure           = np.atleast_1d(p)
+            conditions.freestream.temperature        = np.atleast_1d(T)
+            conditions.freestream.density            = np.atleast_1d(rho)
+            conditions.freestream.dynamic_viscosity  = np.atleast_1d(mu)
+            conditions.freestream.gravity            = np.atleast_1d(9.81)
```
Comment (timdmacdo) [inline trunk/SUAVE/Methods/Propulsion/ramjet_sizing.py]: Maybe these aren't practical to change right now, but these values shouldn't be hard coded.

File: trunk/SUAVE/Methods/Propulsion/rayleigh.py [hunk 17]
```diff
@@ -0,0 +1,56 @@
...
+
+# ----------------------------------------------------------------------
+#  rayleigh
+# ----------------------------------------------------------------------
+
+## @ingroup Methods-Propulsion
+
+
+def rayleigh(gamma, M0, TtR):
+    """
```
Comment (timdmacdo) [inline trunk/SUAVE/Methods/Propulsion/rayleigh.py]: Needs a source
